pkgs/games/badlion-client/default.nix

```
@@ -0,0 +1,32 @@
+{ appimageTools, fetchurl, lib }:
+
+let
+  pname = "badlion-client";
+  version = "3.15.0";
+
+  src = fetchurl {
+    url = "https://client-updates-cdn77.badlion.net/BadlionClient";
+    hash = "sha256-HqMgY9+Xnp4uSTWr//REZGv3p7ivwLX97vxGD5wqu9E=";
+  };
+
+  appimageContents = appimageTools.extract { inherit pname version src; };
+in
+  appimageTools.wrapType2 rec {
+    inherit pname version src;
+
+    extraInstallCommands = ''
+      mv $out/bin/{${pname}-${version},${pname}}
+      install -Dm444 ${appimageContents}/BadlionClient.desktop $out/share/applications/BadlionClient.desktop
+      install -Dm444 ${appimageContents}/BadlionClient.png $out/share/pixmaps/BadlionClient.png
+      substituteInPlace $out/share/applications/BadlionClient.desktop \
+        --replace 'Exec=AppRun --no-sandbox %U' 'Exec=badlion-client'
+    '';
+
+    meta = with lib; {
+      description = "The Most Complete All-In-One Mod Library for Minecraft with 100+ Mods, FPS Improvements, and more.";
+      homepage = "https://client.badlion.net";
+      license = with licenses; [ unfree ];
+      maintainers = with maintainers; [ ometic ];
+      platforms = [ "x86_64-linux" ];
```

I'm not totally sure, but you should be able to change this to `platforms.linux`, which is probably more readable in the long run.
```suggestion
      platforms = platforms.linux;
```